File: C-Travel-Booking-CLI/travel.py
#!/usr/bin/python
#Travel Program using Python

rates = {
1:{
    "min":50,
    "over":55,
    "wait": 100
},
2:{
    "min":60,
    "over":55,
    "wait":200
}
}
def generatebill(pnumber):
    global rates
    min_, over, wait = rates[pnumber].items()
    #print(min_,over,wait)
    dist_trav = 0
    wait_time = 0
    charge_per = 0

    print("\n")
    print("\n")
    print("--------------Enter details below----------------------\n")
    dist_trav = int(input("          --> Enter distance traveled: "))
    wait_time = int(input("          --> Enter waiting hours:"))
    #per km 50 + waiting charge

    if dist_trav < 50:
        charge_per = min_[1] * dist_trav
    elif dist_trav > 50:
        charge_per = over[1] * dist_trav
    if wait_time:
        wait_time *= wait[1]

    total = charge_per + wait_time
    print(f"customer total {total}")
    


def package(package):
    if package == 1:
        print("----------------------------------GOLD package------------------------------\n")
        print("                      |-------------------------------------------|\n")
        print("                      | Charge per KM  |  Waiting charge per hour | \n")
        print("                      |-------------------------------------------|\n")
        print("                      | Rs.50          |  Rs.100                  | \n")
        print("                      |-------------------------------------------|\n")
        print("                      | Note : After the first 50kms the charge per KM is Rs55 | \n")
    elif package==2:
        print("----------------------------------STANDARD package------------------------------\n")
        print("                      |-------------------------------------------|\n")
        print("                      | Charge per KM  |  Waiting charge per hour | \n")
        print("                      |-------------------------------------------|\n")
        print("                      | Rs.60          |  Rs.200                  | \n")
        print("                      |-------------------------------------------|\n")
        print("                      | Note : After the first 40kms the charge per KM is Rs50 | \n")
        print("\n")
    else:
        print("The package you selected does not exist . Please try again\n")
    
    bill = input("DO YOU WANT TO BILL: ")
    if bill in ["yes", "Y", "y"]:
        generatebill(package)
    else:
        print("Exiting")
        return

def selection(option):
    pack = 0
    if option == "B" or option == "b":
        print("Do Your Billings HERE\n")
        print("Billings 1 : GOLD          Billings 2 : STANDARD\n")
    elif option == "P" or option == "p":
        print("You have selected to to view Packages\n")
        print("Package 1 : GOLD        Package 2 :STANDARD \n")
    else:
        print("The BILLING  TPYE you selected does not exist . Please try again\n")
    pack = int(input("Select package: "))
    if pack in (1,2):
        package(pack)
    else:
        print("The PACKAGE you selected does not exist . Please try again\n")


def main():
    print("****|||||||||      HELLO! WELCOME USER    |||||||||****\n")
    print("\n")
    print("---> TO VIEW THE PACKAGES ENTER P/p\n")
    print("---> TO START BILLING ENTER B/b\n")
    opt = input("SELECT YOUR OPTION: ")
    selection(opt)

main()

File: C-Travel-Booking-CLI/travel.c
//Travel Program using C

#include<stdio.h>

//User defined function declaration
void GenerateBill(int package);
float CalculateBill(int distance_travelled ,int waiting_time, int pack);
void ViewPackage(int package);

char decision; // global variable declaration
char true_='y';
void main()
{
    char option='\0';


    printf("****|||||||||      HELLO! WELCOME USER    |||||||||****\n");
    printf("\n");
    printf("\n");
    printf("\n");
    printf("---> TO VIEW THE PACKAGES ENTER P/p\n");
    printf("---> TO START BILLING ENTER B/b\n");
    scanf("%c",&option);

    Selection(option);

}

void Selection(int option)
{
    int package=0;
     if((option=='B') || (option=='b'))
        {
         printf("You have selected to billing\n");
    printf("Package 1 : GOLD          Package 2 : STANDARD\n");
    printf("Select package :");
    scanf("%d",&package);

     //method two using If()
    if ((package==1) || (package==2))
        {
    GenerateBill(package);
        }
    else{
        printf("The package you selected does not exist . Please try again\n");
        }

        }
  else if ((option=='P') || (option=='p'))
    {
    printf("You have selected to to view Packages\n");
    printf("Package 1 : GOLD        Package 2 :STANDARD \n");
    printf("Select package :");
    scanf("%d",&package);

    ViewPackages(package);

    }
}


void ViewPackages(int package)
{
    if (package==1)
        {

      printf("----------------------------------GOLD package------------------------------\n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Charge per KM  |  Waiting charge per hour | \n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Rs.50          |  Rs.100                  | \n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Note : After the first 50kms the charge per KM is Rs55 | \n");

        }
    else if(package==2)
        {
      printf("----------------------------------STANDARD package------------------------------\n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Charge per KM  |  Waiting charge per hour | \n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Rs.45          |  Rs.200                  | \n");
      printf("                      |-------------------------------------------|\n");
      printf("                      | Note : After the first 40kms the charge per KM is Rs50 | \n");
      printf("\n");
        }
    else{
        printf("The package you selected does not exist . Please try again\n");
        }

          printf("Do you want to Generate Bill?(y/n)");
            while(true_=='y')
        {
          scanf("%c",&decision);
    if(decision=='y')
            {
            GenerateBill(package);
            }
    else if(decision=='n')
            {
            main();
            }

        }
}



  void GenerateBill(int selec_package)
  {
    int distance_travelled=0;
    int waiting_time=0;
    float total=0;
     printf("\n");
      printf("\n");
      printf("--------------Enter details below----------------------\n");
      printf("          --> Enter distance traveled: ");
      scanf("%d",&distance_travelled);

      printf("          --> Enter waiting hours:");
      scanf("%d",&waiting_time);

      total = CalculateBill(distance_travelled,waiting_time, selec_package);
      printf("Customer Total : Rs.%f",total);
       printf("\n");
        printf("\n");
          printf("Do you want to Generate Bill?(y/n)");
        while(true_=='y')
        {
          scanf("%c",&decision);
            if(decision=='y')
                    {
                    Selection('B');
                    }
            else if(decision=='n')
                    {
                    main();
                    }
        }


  }


  float CalculateBill(int distance_travelled ,int waiting_time, int pack)
  {
    int km_charge=0;
    int waiting_charge=0;
    float total_=0;
    if(distance_travelled<=0)
    {
        printf("Distance traveled cannot be less than or 0\n");
        printf("Do you want to re-try Billing?(y/n)");
        while(true_=='y')
        {
          scanf("%c",&decision);
            if(decision=='y')
                    {
                    GenerateBill(pack);
                    }
            else if(decision=='n')
                    {
                    main();
                    }
        }
    }
    else
        {
          if (pack ==1)
    {
        waiting_charge=100;
        if (distance_travelled>50)
        {
        km_charge=55;

        }
       else if(distance_travelled<50)
       {
        km_charge=50;

       }

int km_charge=0;
    int waiting_charge=0;
    float total_=0;
    }

     else if (pack ==2)
    {
        waiting_charge=200;
        if (distance_travelled>40)
        {
        km_charge=50;

        }
       else if(distance_travelled<40)
       {
        km_charge=45;

       }
        total_ =  (distance_travelled * km_charge) + (waiting_time * waiting_charge);
    }
        }



return total_;
  }
